### PR TITLE
`random_generalized_svd` dostring fixes

### PR DESCRIPTION
**Pull request by artpelling**
This PR will fix some typos in the docstring and logger of `random_generalized_svd`. I also had a read of the cited paper and I think the new docstring is easier to understand.

### CODE REVIEW

File: src/pymor/algorithms/rand_la.py [hunk 1]
```diff
@@ -170,14 +170,24 @@ def rrf(A, source_product=None, range_product=None, q=2, l=8, return_rand=False,
...
     .. math::
 
-        A = U \Sigma V^H \mathtt{source_product}
+        A = U \Sigma V^{-1},
+
+    where the inner product on the range :math:`\mathbb{R}^m` is given by
+
+    .. math::
+
+        \langle x, y \rangle_S = x^TSy
```
Comment (sdrave) [inline src/pymor/algorithms/rand_la.py:184]:
```suggestion
        (x, y)_S = x^TSy
```

This is the notation, I have used in the 'Building a Reduced Basis' tutorial, and I'd say it's also more standard in mathematics. Pointy brackets are for dual pairings ...

File: src/pymor/algorithms/rand_la.py [hunk 2]
```diff
@@ -170,14 +170,24 @@ def rrf(A, source_product=None, range_product=None, q=2, l=8, return_rand=False,
...
+
+    and the inner product on the source :math:`\mathbb{R}^n` is given by
+
+    .. math::
+
+        \langle x, y \rangle_T = x^TTy.
```
Comment (sdrave) [inline src/pymor/algorithms/rand_la.py:190]:
```suggestion
        (x, y) = x^TTy.
```
see above.

**Review by sdrave — CHANGES_REQUESTED** [review]

**Comment by artpelling** [comment]
Thanks! I had swapped the notation too many times :sweat_smile:

**Review by sdrave — APPROVED** [review]

**Comment by sdrave** [comment]
@artpelling, I think you need to rebase on main to make ci pass.

**Comment by sdrave** [comment]
This should also be changed in `adaptive_rrf` then. @pmli, do you see any reasons not to use the default tolerances?

**Comment by renefritze** [comment]
The problem actually was that Github Actions aren't run for a commit (The conda env update) added from Github Actions. I've reordered the commits here with a rebase.

**Comment by sdrave** [comment]
@artpelling, can you rebase again? Hopefull, ci should finally pass.

**Comment by renefritze** [comment]
> @artpelling, can you rebase again?

Went ahead and did that

**Comment by sdrave** [comment]
I think another rebase is necessary. The hypothesis fixes hadn't been merged yet.

**Pull request merged**